tests/kokkos/kokkos_utils_tests.cpp

```diff
@@ -244,4 +245,81 @@ TEST_CASE("parallel_reduce", "[kokkos_utils]")
   test_parallel_reduce<Real,15,false> ();
 }
 
+
+template<typename Scalar, int TotalSize, int VectorSize, bool Serialize>
+void test_view_reduction()
+{
+  using Device = ekat::DefaultDevice;
+  using MemberType = typename ekat::KokkosTypes<Device>::MemberType;
+  using ExeSpace = typename ekat::KokkosTypes<Device>::ExeSpace;
+
+  ekat::ExeSpaceUtils<Serialize,ExeSpace> ExeSpaceUtils; 
+  
+  using PackType = ekat::Pack<Scalar, VectorSize>;
+  using ViewType = Kokkos::View<PackType*,ExeSpace>;
+
+  const int view_length = (int)std::ceil(1.0*TotalSize/VectorSize);
+
+  Scalar serial_result = Scalar();
```

Comment:
Since your impl supports it, it would be good to test the case where the result already contains an initial value, to make sure it is not overwritten. Could be a random value, or some FP number not representable exactly (to avoid surprises), like 1.0/3.0.

Reply:
The implementation of the pack `reduce_sum()` supports it, but the implementation of `view_reduction()` does not. I could implement this by adding a `ValueType a` input like in the pack reduce function.

Reply:
I tested this and, in the implementation for both Serial and Parallel, `result` gets overwritten. I could implement  such that it does not get overwritten if you like?

Reply:
Ah, I got confused with `reduce_sum`. Well, since you impl-ed it for `reduce_sum`, it would make sense to impl it also here, no? Same logic should work  here.

Reply:
I changed the implementation to add the reduction to whatever value is stored in result.